File: frontend/src/types/quiz.ts
export interface QuizQuestion {
    question: string;
    options: string[];
    correctAnswer: number;
    explanation: string;
  }
  
  export interface QuizRequest {
    text: string;
    num_questions?: number;
    difficulty?: string;
  }
  
  export interface QuizResponse {
    questions: QuizQuestion[];
  }

File: backend/app/models/quiz.py
from pydantic import BaseModel
from typing import List, Optional

class QuizRequest(BaseModel):
    text: str
    num_questions: int = 5
    difficulty: Optional[str] = "medium"

class Question(BaseModel):
    question: str
    options: List[str]
    correct_answer: int
    explanation: str

class QuizResponse(BaseModel):
    questions: List[Question]